internal/table/table.go

```diff
@@ -0,0 +1,167 @@
+package table
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/ActiveState/cli/internal/logging"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+const dash = "\u2500"
+
+type FormatFunc func(string, ...interface{}) string
+
+type Table struct {
+	rows   [][]string
+	header []string
+	// TODO: Make column width part of a row struct?
+	columnWidths []int
+	padding      int
+	// TODO: Add logic to deal with max column size
+	maxSize int
```

Comment:
These seem like values that should be passed through functions rather than recorded to the struct